src/odemis/acq/stream/_sync.py

```diff
@@ -792,57 +824,56 @@ def _onCompletedData(self, n, raw_das):
         raw_das (list of DataArray): data as received from the detector.
            The data is ordered, with X changing fast, then Y slow
         """
+
         # Default is to assume the data is 2D and assemble it.
         da = self._assembleTiles(self.repetition.value, raw_das)
         # explicitly add names to make sure they are different
         da.metadata[MD_DESCRIPTION] = self._streams[n].name.value
+
         self._raw.append(da)
 
     def _runAcquisition(self, future):
         """
         Acquires images from the multiple detectors via software synchronisation.
+        Select whether the ebeam is moved for scanning or the sample stage.
+        """
+
+        if hasattr(self, "useScanStage") and self.useScanStage.value:
+            return self._runAcquisitionScanStage(future)
+        else:
+            return self._runAcquisitionEbeam(future)
+
+    def _runAcquisitionEbeam(self, future):
+        """
+        Acquires images from the multiple detectors via software synchronisation.
+        Acquires images via moving the ebeam.
         Warning: can be quite memory consuming if the grid is big
         returns (list of DataArray): all the data acquired
         raises:
           CancelledError() if cancelled
           Exceptions if error
         """
-        if hasattr(self, "useScanStage") and self.useScanStage.value:
-            return self._runAcquisitionScanStage(future)
-
         # TODO: handle better very large grid acquisition (than memory oops)
         try:
             self._acq_done.clear()
             rep_time = self._adjustHardwareSettings()
             dwell_time = self._emitter.dwellTime.value
             sem_time = dwell_time * numpy.prod(self._emitter.resolution.value)
-            spot_pos = self._getSpotPositions()
+            spot_pos = self._getSpotPositions()  # list of positions of ebeam spots
```

Comment:
In general, if the function already has a description of what it returns and you directly store it, it's not necessary to repeat it in the caller. If you really like to have a comment, change it to something like: `list of center positions for each point of the ROI`. It's clearer that sub-pixels positions are not computed.